scripts/train/train.py

```diff
@@ -510,11 +510,16 @@ def main(cfg: DictConfig) -> Trainer:
 
     # Dataloaders
     log.info('Building train loader...')
-    train_loader = build_dataloader(
-        train_loader_config,
-        tokenizer,
-        device_train_batch_size,
-    )
+    try:
```

Comment:
Instead of doing this try-except, wdyt about using sys.excepthook with @dakinggg's new custom init code extension: https://github.com/mosaicml/llm-foundry/blob/main/scripts/train/train.py#L163

Reply:
I realize @dakinggg's init code extension point won't work with the data prep scripts. Is it still possible to use sys.excepthook directly instead of try-except?

Reply:
I'm planning to do try-except as a temp solution (~next month or so). Another option would be to create an excepthook override script in foundry and run it at the start of `train.py` + all the data prep scripts. 

I am working on a cloud change to inject excepthook override for all python interpreters used during a run, and then we can remove these.

Reply:
Gotcha, OK with leaving the try-excepts as is then. Thanks for explaining!